boa/src/builtins/number/tests.rs

```diff
@@ -677,3 +677,178 @@ fn parse_float_too_many_args() {
 
     assert_eq!(&forward(&mut engine, "parseFloat(\"100.5\", 10)"), "100.5");
 }
+
+#[test]
+fn global_is_finite() {
+    let realm = Realm::create();
+    let mut engine = Interpreter::new(realm);
+
+    assert_eq!("false", &forward(&mut engine, "isFinite(Infinity)"));
+    assert_eq!("false", &forward(&mut engine, "isFinite(NaN)"));
+    assert_eq!("false", &forward(&mut engine, "isFinite(-Infinity)"));
+    assert_eq!("true", &forward(&mut engine, "isFinite(0)"));
+    assert_eq!("true", &forward(&mut engine, "isFinite(2e64)"));
+    assert_eq!("true", &forward(&mut engine, "isFinite(910)"));
+    assert_eq!("true", &forward(&mut engine, "isFinite(null)"));
+    assert_eq!("true", &forward(&mut engine, "isFinite('0')"));
+    assert_eq!("false", &forward(&mut engine, "isFinite()"));
+}
+
+#[test]
+fn global_is_nan() {
+    let realm = Realm::create();
+    let mut engine = Interpreter::new(realm);
+
+    assert_eq!("true", &forward(&mut engine, "isNaN(NaN)"));
+    assert_eq!("true", &forward(&mut engine, "isNaN('NaN')"));
+    assert_eq!("true", &forward(&mut engine, "isNaN(undefined)"));
+    assert_eq!("true", &forward(&mut engine, "isNaN({})"));
+    assert_eq!("false", &forward(&mut engine, "isNaN(true)"));
+    assert_eq!("false", &forward(&mut engine, "isNaN(null)"));
+    assert_eq!("false", &forward(&mut engine, "isNaN(37)"));
+    assert_eq!("false", &forward(&mut engine, "isNaN('37')"));
+    assert_eq!("false", &forward(&mut engine, "isNaN('37.37')"));
+    assert_eq!("true", &forward(&mut engine, "isNaN('37,5')"));
+    assert_eq!("true", &forward(&mut engine, "isNaN('123ABC')"));
+    // Incorrect due to ToNumber implementation inconsistencies.
```

Comment:
String to number with the `to_number` function is not 100% spec compliant: <https://github.com/boa-dev/boa/blob/master/boa/src/exec/mod.rs#L329>

To fix this we should trim and if it is a empty string return 0 else we parse as we currently do now.

Reply:
Should this be done in this PR or should there be a separate one to try and make to_number fully spec compliant?

Reply:
> Should this be done in this PR or should there be a separate one to try and make to_number fully spec compliant?

Probably in a separate PR is better :)